File: scripts/format_munge.py
import pandas as pd
import numpy as np

df = pd.read_csv(snakemake.input["GWAS"], delim_whitespace=True)

df["Z"] = np.sqrt(df.Stat)*df.Direction

df = df.rename(columns = {"Predictor": "SNP", "n": "N"})

df[["SNP", "A1", "A2", "Z", "N"]].to_csv(snakemake.output[0], sep = '\t', index = False)

File: scripts/format_munge.R
df = read.table(snakemake@input[["GWAS"]], header = T)
snp_df = read.csv(snakemake@input[["snp_map"]], sep = '\t')

df = merge(df, snp_df, by.x = "Predictor", by.y = "id")

df$Z = sqrt(df$Stat)*df$Direction
df$p = 2*pnorm(abs(df$Z), lower.tail = F)

df = df[, c("rsid", "A1", "A2", "Z", "n", "p")]
names(df) = c("SNP", "A1", "A2", "Z", "N", "p")

write.table(df, snakemake@output[[1]], sep = '\t', row.names = F, quote = F)
